airflow-core/src/airflow/models/dag.py

```diff
@@ -580,7 +580,7 @@ def deactivate_deleted_dags(
                 dm.is_stale = True
 
     @classmethod
-    def dags_needing_dagruns(cls, session: Session) -> tuple[Query, dict[str, datetime]]:
+    def dags_needing_dagruns(cls, session: Session) -> tuple[ScalarResult[Any], dict[str, datetime]]:
```

Comment:
Can we annotate this as ScalarResult[DagModel] instead? Same for other things below, it would e much better if they annotate what type is being returned (instead of Any)